spec/models/bag_man_request_spec.rb

```diff
@@ -59,31 +59,42 @@
     end
 
     context "when not cancelled" do
+      let(:replication) {
+        mock = double(:replication_transfer)
+        allow(mock).to receive(:cancel!)
+        allow(mock).to receive(:from_node).and_return FromNode.new("fake_namespace")
+        mock
+      }
+      let(:request) {
+        request = Fabricate(:bag_man_request)
+        allow(request).to receive(:replication_transfer).and_return(replication)
+        request
+      }
       before(:each) do
-        @repl_mock = double(:replication_transfer)
-        allow(@repl_mock).to receive(:cancel!)
-        allow(@repl_mock).to receive(:from_node).and_return FromNode.new("fake_namespace")
-        @request = Fabricate(:bag_man_request)
-        allow(@request).to receive(:replication_transfer).and_return(@repl_mock)
         allow(Client::Repl::CancelJob).to receive(:perform_later)
       end
       it "sets the cancel_reason" do
-        @request.cancel!('testing')
-        expect(@request.reload.cancel_reason).to eql('testing')
+        request.cancel!('testing')
+        expect(request.reload.cancel_reason).to eql('testing')
       end
```

Comment:
It might be shortened to:
```ruby
      it "sets the cancel_reason" do
        expect(request.cancel_reason).not_to eql('testing')
        change { request.cancel!('testing') }.to eql('testing')
      end
```